### PR TITLE
Use the natively supported version and usage by picocli

### PR DESCRIPTION
**Pull request by jonasgeiregat**
Instructions:
1. The PR has to be tagged with at least one of the following labels (*):
   1. `feature`
   2. `bugfix`
2. Remove these instructions before publishing the PR.
 
(*) Other labels to consider:
- `testing`
- `refactor`
- `cleanup`

The general idea was to make use of the natively supported `mixinStandardHelpOptions` of picocli.

I took the liberty to add some tests for the `PinotAdministrator`.

Amy feedback more then welcome I'm happy to adjust the code in order to get it merged in and improve pinot!

### CODE REVIEW

**Review by walterddr — COMMENTED** [review]
thank you for the PR, can you clean up the linter issue and submit again.

**Comment by jonasgeiregat** [comment]
> thank you for the PR, can you clean up the linter issue and submit again.

I ran checkstyle and fixed all open issues.

**Comment by jonasgeiregat** [comment]
> thank you for the PR, can you clean up the linter issue and submit again.

There seem to be more then checkstyle in place. I suppose I need to get the built green. I'll look into it.

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/PinotAdministrator.java [hunk 1]
```diff
@@ -140,56 +142,71 @@ public class PinotAdministrator {
...
-      if (!parseResult.hasSubcommand()) {
-        if (_version) {
-          printVersion();
-          _status = 0;
-        } else if (_help) {
-          printUsage();
-          _status = 0;
-        }
-      } else {
+      var commandLine = new CommandLine(this);
```
Comment (walterddr) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/PinotAdministrator.java:167]: seems like we can't support var b/c of legacy JDK8 support :-(

Comment (jonasgeiregat) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/PinotAdministrator.java]: I wasn't aware my apologies. I removed all JDK8+ specific things.

Comment (walterddr) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/PinotAdministrator.java]: no worries. i wasn't aware either. thank you for quick turn around!

**Comment by jonasgeiregat** [comment]
Saw an attempt was made to built the project but, failed due to incompatibility with java 8.

Think I forgot to actually push my changes 🙈 

In any case it should now pass, can you try again ?

Also what is the reason for supporting JDK-8 as a minimum version ?

**Comment by walterddr** [comment]
> Also what is the reason for supporting JDK-8 as a minimum version ?

I think it was because some users were still on JDK8. we are in the process of removing it (for a while unfortunately)
I will take another review later today. thank you for the contribution

File: pinot-tools/src/test/java/org/apache/pinot/tools/admin/PinotAdministratorTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.tools.admin;
+
+import java.io.StringWriter;
```
Comment (walterddr) [inline pinot-tools/src/test/java/org/apache/pinot/tools/admin/PinotAdministratorTest.java:21]:
```suggestion
import com.google.common.collect.ImmutableMap;
import java.io.StringWriter;
```

File: pinot-tools/src/test/java/org/apache/pinot/tools/admin/PinotAdministratorTest.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+    // then
+    assertEquals(out.toString(), "List All Pinot Component Versions:\n" + "Package: AddSchema, Version: 1.2.3\n"
+                                 + "Package: AddTable, Version: 1.2.4\n");
+  }
+
+  static class VersionsStub extends Versions {
+    @Override
+    public Map<String, String> getComponentVersions() {
+      return new TreeMap<>(Map.of("AddSchema", "1.2.3", "AddTable", "1.2.4"));
```
Comment (walterddr) [inline pinot-tools/src/test/java/org/apache/pinot/tools/admin/PinotAdministratorTest.java:167]:
```suggestion
      return new TreeMap<>(ImmutableMap.of("AddSchema", "1.2.3", "AddTable", "1.2.4"));
```

File: pinot-tools/src/test/java/org/apache/pinot/tools/admin/PinotAdministratorTest.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  public void testHelpParameterShouldOutputAllPossibleParametersAndTheirDescriptions() {
+    // given
+    StringWriter out = new StringWriter();
+    PinotAdministrator pinotAdministrator = new PinotAdministrator(out, new Versions());
+
+    // when
+    pinotAdministrator.execute(new String[]{"AddSchema", "--help"});
+
+    // then
+    assertEquals(out.toString(), "Usage: <main class> AddSchema [-hV] [-exec] [-authToken=<_authToken>]\n"
```
Comment (walterddr) [inline pinot-tools/src/test/java/org/apache/pinot/tools/admin/PinotAdministratorTest.java:42]: this assertEquals is going to be hard to maintain if we add additional commands. was wondering how other project use picocli handles unit-test? do we have a util to generate these expected results?

File: pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java [hunk 5]
```diff
@@ -107,53 +109,37 @@ private enum QueryMode {
...
 
-  @Override
-  public String description() {
-    return "Run queries from a query file in singleThread, multiThreads, targetQPS or increasingQPS mode. E.g.\n"
-        + "  QueryRunner -mode singleThread -queryFile <queryFile> -numTimesToRunQueries 0 "
-        + "-numIntervalsToReportAndClearStatistics 5\n"
-        + "  QueryRunner -mode multiThreads -queryFile <queryFile> -numThreads 10 -reportIntervalMs 1000\n"
-        + "  QueryRunner -mode targetQPS -queryFile <queryFile> -startQPS 50\n"
-        + "  QueryRunner -mode increasingQPS -queryFile <queryFile> -startQPS 50 -deltaQPS 10 "
-        + "-numIntervalsToIncreaseQPS 20\n";
```
Comment (walterddr) [inline pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java:128]: did we missed these examples in the new format?

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/filesystem/MoveFiles.java [hunk 6]
```diff
@@ -81,13 +82,8 @@ public boolean execute()
...
-  }
-
-  @Override
-  public String description() {
-    return "Move file from source to destination, moving from two different PinotFS is not supported.";
-  }
+//  @Override
+//  public void getDescription() {
+//    System.out.println("mv [-o] <source-uri> <destination-uri>");
+//  }
```
Comment (walterddr) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/filesystem/MoveFiles.java:88]: remove these comments if no longer needed

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/RebalanceTableCommand.java [hunk 7]
```diff
@@ -111,82 +105,4 @@ public boolean execute()
...
     return rebalanceResult.getStatus() == RebalanceResult.Status.DONE;
   }
-
-  @Override
-  public String description() {
-    return "Reassign instances and segments for a table.";
-  }
-
-  @Override
-  public void printExamples() {
```
Comment (walterddr) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/RebalanceTableCommand.java:121]: is this description refactored anywhere in the new code? can't find it

**Review by walterddr — COMMENTED** [review]
did a brief pass through the PR. please kindly take a look at my comments. 

thank you for addressing the JDK8 issue. it is tedious

**Pull request closed without merging**